pkg/controller/core/workload_controller_test.go

```diff
@@ -1314,6 +1314,95 @@ func TestReconcile(t *testing.T) {
 				}).
 				Obj(),
 		},
+		"shouldn't delete the workflow because, object retention not configured": {
+			workload: utiltesting.MakeWorkload("wl", "ns").
+				Condition(metav1.Condition{
+					Type:   kueue.WorkloadFinished,
+					Status: metav1.ConditionTrue,
+				}).
+				Obj(),
+			reconcilerOpts: []Option{
+				WithWorkloadObjectRetention(nil),
+			},
+			wantWorkload: utiltesting.MakeWorkload("wl", "ns").
+				Condition(metav1.Condition{
+					Type:   kueue.WorkloadFinished,
+					Status: metav1.ConditionTrue,
+				}).
+				Obj(),
+			wantError: nil,
+		},
+		"shouldn't try to delete the workload (no event emitted) because it is already being deleted by kubernetes, object retention configured": {
+			workload: utiltesting.MakeWorkload("wl", "ns").
+				Condition(metav1.Condition{
+					Type:   kueue.WorkloadFinished,
+					Status: metav1.ConditionTrue,
+					LastTransitionTime: metav1.Time{
+						Time: testStartTime,
+					},
+				}).
+				DeletionTimestamp(testStartTime).
+				Finalizers(kueue.ResourceInUseFinalizerName).
+				Obj(),
+			reconcilerOpts: []Option{
+				WithWorkloadObjectRetention(&metav1.Duration{
+					Duration: 24 * time.Hour,
+				}),
+			},
+			wantWorkload: nil,
+			wantError: nil,
+		},
+		"shouldn't try to delete the workload because the retention period hasn't elapsed yet, object retention configured": {
+			workload: utiltesting.MakeWorkload("wl", "ns").
+				Condition(metav1.Condition{
+					Type: kueue.WorkloadFinished,
+					Status: metav1.ConditionTrue,
+					LastTransitionTime: metav1.NewTime(testStartTime.Add(-10 * time.Second)),
+				}).
+				Obj(),
+			reconcilerOpts: []Option{
+				WithWorkloadObjectRetention(
+					&metav1.Duration{
+						Duration: 30 * time.Second,
+				}),
+			},
+			wantWorkload: utiltesting.MakeWorkload("wl", "ns").
+				Condition(metav1.Condition{
+					Type: kueue.WorkloadFinished,
+					Status: metav1.ConditionTrue,
+					LastTransitionTime: metav1.NewTime(testStartTime.Add(-10 * time.Second)),
+				}).
+				Obj(),
+			wantError: nil,
+		},
+		"should delete the workload because the retention period has elapsed, object retention configured": {
```

Comment:
Can we add a test case for a workload with finalizer? I would like to also consider an integration test for it.

Reply:
Sure things. It'd greatly help me to understand the case if you could respond to my comment above first about cleaning up Workloads with a finalizer 🙏